samples/nrf9160/tcp_client/src/main.c

```diff
@@ -0,0 +1,230 @@
+
+		len = snprintf(log_buf, sizeof(log_buf),
+			       "eDRX parameter update: eDRX: %f, PTW: %f\n",
+			       evt->edrx_cfg.edrx, evt->edrx_cfg.ptw);
+		if (len > 0) {
+			printk("%s\n", log_buf);
+		}
+		break;
+	}
+	case LTE_LC_EVT_RRC_UPDATE:
+		printk("RRC mode: %s\n",
+			evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ?
+			"Connected" : "Idle\n");
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		printk("LTE cell changed: Cell ID: %d, Tracking area: %d\n",
+		       evt->cell.id, evt->cell.tac);
+		break;
+	default:
+		break;
+	}
+}
+
+static int configure_low_power(void)
+{
+	int err;
+
+#if defined(CONFIG_TCP_PSM_ENABLE)
+	/** Power Saving Mode */
+	err = lte_lc_psm_req(true);
+	if (err) {
+		printk("lte_lc_psm_req, error: %d\n", err);
+	}
+#else
+	err = lte_lc_psm_req(false);
+	if (err) {
+		printk("lte_lc_psm_req, error: %d\n", err);
+	}
+#endif
+
+#if defined(CONFIG_TCP_EDRX_ENABLE)
+	/** enhanced Discontinuous Reception */
+	err = lte_lc_edrx_req(true);
+	if (err) {
+		printk("lte_lc_edrx_req, error: %d\n", err);
+	}
+#else
+	err = lte_lc_edrx_req(false);
+	if (err) {
+		printk("lte_lc_edrx_req, error: %d\n", err);
+	}
+#endif
+
+#if defined(CONFIG_TCP_RAI_ENABLE)
+	/** Release Assistance Indication  */
+	err = lte_lc_rai_req(true);
+	if (err) {
+		printk("lte_lc_rai_req, error: %d\n", err);
+	}
+#endif
+
+	return err;
+}
+
+static void modem_configure(void)
+{
+	int err;
+
+	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
+		/* Do nothing, modem is already configured and LTE connected. */
+	} else {
+		err = lte_lc_init_and_connect_async(lte_handler);
+		if (err) {
+			printk("Modem configuration, error: %d\n", err);
+			return;
+		}
+	}
+}
+#endif
+
+static void server_disconnect(void)
+{
+	(void)close(client_fd);
+}
+
+static int server_init(void)
+{
+	struct sockaddr_in *server4 = ((struct sockaddr_in *)&host_addr);
+
+	server4->sin_family = AF_INET;
+	server4->sin_port = htons(CONFIG_TCP_SERVER_PORT);
+
+	inet_pton(AF_INET, CONFIG_TCP_SERVER_ADDRESS_STATIC,
+		  &server4->sin_addr);
+
+	return 0;
+}
+
+static int server_connect(void)
+{
+	int err;
+
+	client_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	if (client_fd < 0) {
+		printk("Failed to create TCP socket: %d\n", errno);
+		goto error;
+	}
+
+	err = connect(client_fd, (struct sockaddr *)&host_addr,
+		      sizeof(struct sockaddr_in));
+	if (err < 0) {
+		printk("Connect failed : %d\n", errno);
+		goto error;
+	} else {
+		printk("Successfully connected to TCP server: %s on port %d\n",
+		CONFIG_TCP_SERVER_ADDRESS_STATIC, CONFIG_TCP_SERVER_PORT);
+	}
+
+	return 0;
+
+error:
+	server_disconnect();
+
+	return err;
+}
+
+void main(void)
+{
+	int err;
+
+	printk("TCP Client sample has started\n");
+
+	work_init();
+
+#if defined(CONFIG_BSD_LIBRARY)
+	err = configure_low_power();
+	if (err) {
+		printk("Unable to set low power configuration, error: %d\n",
+		       err);
+	}
+
+	modem_configure();
+
+	k_sem_take(&lte_connected, K_FOREVER);
+#endif
+
+	err = server_init();
```

Comment:
Why are these functions called `server_`, aren't we the client after all?

Reply:
hehe, I agree. 
Could be a good idea to change that in the udp sample as well 👍